core/integration/module_test.go

```diff
@@ -1119,6 +1119,102 @@ func (m *Minimal) HelloFinal(
 	require.Equal(t, "", prop.Get("description").String())
 }
 
+func TestModuleGoWeirdFields(t *testing.T) {
+	// these are all cases that used to panic due to the disparity in the type spec and the ast
+
+	t.Parallel()
+
+	c, ctx := connect(t)
+
+	modGen := c.Container().From(golangImage).
+		WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+		WithWorkdir("/work").
+		With(daggerExec("mod", "init", "--name=minimal", "--sdk=go")).
+		WithNewFile("main.go", dagger.ContainerWithNewFileOpts{
+			Contents: `package main
+
+type Z string
+
+type Minimal struct {
+	// field with single (normal) name
+	W string
+
+	// field with multiple names
+	X, Y string
+
+	// field with no names
+	Z
```

Comment:
So. Why `type Z string` instead of `type Z struct{}`?

Well, this actually doesn't work at the moment - when you embed structs in go, the json encoder marshals/unmarshals this by lifting all of those properties to the top-level.

But, in our field traversal, we don't actually do this step, so we declare that we have a `Z` field on `Minimal` - but the json encoder will never create this field, and so we consistently fail with "cannot be nullable" errors.

The right solution for this is to handle the struct embedding case, but this is *fiddly* - and not actually related to how we're doing the field names, which is the subject of this PR. I'll follow up with another patch for this.